File: src/main/java-graphql/io/github/graphql/model/GistResponseProjection.java
package io.github.graphql.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField;
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection;

/**
 * Response projection for Gist
 */
@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:47+0800"
)
public class GistResponseProjection extends GraphQLResponseProjection {

    public GistResponseProjection() {
    }

    @Override
    public GistResponseProjection all$() {
        return all$(3);
    }

    @Override
    public GistResponseProjection all$(int maxDepth) {
        if (projectionDepthOnFields.getOrDefault("GistResponseProjection.GistCommentConnectionResponseProjection.comments", 0) <= maxDepth) {
            projectionDepthOnFields.put("GistResponseProjection.GistCommentConnectionResponseProjection.comments", projectionDepthOnFields.getOrDefault("GistResponseProjection.GistCommentConnectionResponseProjection.comments", 0) + 1);
            this.comments(new GistCommentConnectionResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("GistResponseProjection.GistCommentConnectionResponseProjection.comments", 0)));
        }
        this.createdAt();
        this.description();
        if (projectionDepthOnFields.getOrDefault("GistResponseProjection.GistFileResponseProjection.files", 0) <= maxDepth) {
            projectionDepthOnFields.put("GistResponseProjection.GistFileResponseProjection.files", projectionDepthOnFields.getOrDefault("GistResponseProjection.GistFileResponseProjection.files", 0) + 1);
            this.files(new GistFileResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("GistResponseProjection.GistFileResponseProjection.files", 0)));
        }
        if (projectionDepthOnFields.getOrDefault("GistResponseProjection.GistConnectionResponseProjection.forks", 0) <= maxDepth) {
            projectionDepthOnFields.put("GistResponseProjection.GistConnectionResponseProjection.forks", projectionDepthOnFields.getOrDefault("GistResponseProjection.GistConnectionResponseProjection.forks", 0) + 1);
            this.forks(new GistConnectionResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("GistResponseProjection.GistConnectionResponseProjection.forks", 0)));
        }
        this.id();
        this.isFork();
        this.isPublic();
        this.name();
        if (projectionDepthOnFields.getOrDefault("GistResponseProjection.RepositoryOwnerResponseProjection.owner", 0) <= maxDepth) {
            projectionDepthOnFields.put("GistResponseProjection.RepositoryOwnerResponseProjection.owner", projectionDepthOnFields.getOrDefault("GistResponseProjection.RepositoryOwnerResponseProjection.owner", 0) + 1);
            this.owner(new RepositoryOwnerResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("GistResponseProjection.RepositoryOwnerResponseProjection.owner", 0)));
        }
        this.pushedAt();
        this.resourcePath();
        if (projectionDepthOnFields.getOrDefault("GistResponseProjection.StargazerConnectionResponseProjection.stargazers", 0) <= maxDepth) {
            projectionDepthOnFields.put("GistResponseProjection.StargazerConnectionResponseProjection.stargazers", projectionDepthOnFields.getOrDefault("GistResponseProjection.StargazerConnectionResponseProjection.stargazers", 0) + 1);
            this.stargazers(new StargazerConnectionResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("GistResponseProjection.StargazerConnectionResponseProjection.stargazers", 0)));
        }
        this.updatedAt();
        this.url();
        this.viewerHasStarred();
        this.typename();
        return this;
    }

    public GistResponseProjection comments(GistCommentConnectionResponseProjection subProjection) {
        return comments((String)null, subProjection);
    }

    public GistResponseProjection comments(String alias, GistCommentConnectionResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("comments").alias(alias).projection(subProjection));
        return this;
    }

    public GistResponseProjection comments(GistCommentsParametrizedInput input, GistCommentConnectionResponseProjection subProjection) {
        return comments(null, input, subProjection);
    }

    public GistResponseProjection comments(String alias, GistCommentsParametrizedInput input, GistCommentConnectionResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("comments").alias(alias).parameters(input).projection(subProjection));
        return this;
    }

    public GistResponseProjection createdAt() {
        return createdAt(null);
    }

    public GistResponseProjection createdAt(String alias) {
        fields.add(new GraphQLResponseField("createdAt").alias(alias));
        return this;
    }

    public GistResponseProjection description() {
        return description(null);
    }

    public GistResponseProjection description(String alias) {
        fields.add(new GraphQLResponseField("description").alias(alias));
        return this;
    }

    public GistResponseProjection files(GistFileResponseProjection subProjection) {
        return files((String)null, subProjection);
    }

    public GistResponseProjection files(String alias, GistFileResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("files").alias(alias).projection(subProjection));
        return this;
    }

    public GistResponseProjection files(GistFilesParametrizedInput input, GistFileResponseProjection subProjection) {
        return files(null, input, subProjection);
    }

    public GistResponseProjection files(String alias, GistFilesParametrizedInput input, GistFileResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("files").alias(alias).parameters(input).projection(subProjection));
        return this;
    }

    public GistResponseProjection forks(GistConnectionResponseProjection subProjection) {
        return forks((String)null, subProjection);
    }

    public GistResponseProjection forks(String alias, GistConnectionResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("forks").alias(alias).projection(subProjection));
        return this;
    }

    public GistResponseProjection forks(GistForksParametrizedInput input, GistConnectionResponseProjection subProjection) {
        return forks(null, input, subProjection);
    }

    public GistResponseProjection forks(String alias, GistForksParametrizedInput input, GistConnectionResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("forks").alias(alias).parameters(input).projection(subProjection));
        return this;
    }

    public GistResponseProjection id() {
        return id(null);
    }

    public GistResponseProjection id(String alias) {
        fields.add(new GraphQLResponseField("id").alias(alias));
        return this;
    }

    public GistResponseProjection isFork() {
        return isFork(null);
    }

    public GistResponseProjection isFork(String alias) {
        fields.add(new GraphQLResponseField("isFork").alias(alias));
        return this;
    }

    public GistResponseProjection isPublic() {
        return isPublic(null);
    }

    public GistResponseProjection isPublic(String alias) {
        fields.add(new GraphQLResponseField("isPublic").alias(alias));
        return this;
    }

    public GistResponseProjection name() {
        return name(null);
    }

    public GistResponseProjection name(String alias) {
        fields.add(new GraphQLResponseField("name").alias(alias));
        return this;
    }

    public GistResponseProjection owner(RepositoryOwnerResponseProjection subProjection) {
        return owner(null, subProjection);
    }

    public GistResponseProjection owner(String alias, RepositoryOwnerResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("owner").alias(alias).projection(subProjection));
        return this;
    }

    public GistResponseProjection pushedAt() {
        return pushedAt(null);
    }

    public GistResponseProjection pushedAt(String alias) {
        fields.add(new GraphQLResponseField("pushedAt").alias(alias));
        return this;
    }

    public GistResponseProjection resourcePath() {
        return resourcePath(null);
    }

    public GistResponseProjection resourcePath(String alias) {
        fields.add(new GraphQLResponseField("resourcePath").alias(alias));
        return this;
    }

    public GistResponseProjection stargazers(StargazerConnectionResponseProjection subProjection) {
        return stargazers((String)null, subProjection);
    }

    public GistResponseProjection stargazers(String alias, StargazerConnectionResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("stargazers").alias(alias).projection(subProjection));
        return this;
    }

    public GistResponseProjection stargazers(GistStargazersParametrizedInput input, StargazerConnectionResponseProjection subProjection) {
        return stargazers(null, input, subProjection);
    }

    public GistResponseProjection stargazers(String alias, GistStargazersParametrizedInput input, StargazerConnectionResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("stargazers").alias(alias).parameters(input).projection(subProjection));
        return this;
    }

    public GistResponseProjection updatedAt() {
        return updatedAt(null);
    }

    public GistResponseProjection updatedAt(String alias) {
        fields.add(new GraphQLResponseField("updatedAt").alias(alias));
        return this;
    }

    public GistResponseProjection url() {
        return url(null);
    }

    public GistResponseProjection url(String alias) {
        fields.add(new GraphQLResponseField("url").alias(alias));
        return this;
    }

    public GistResponseProjection viewerHasStarred() {
        return viewerHasStarred(null);
    }

    public GistResponseProjection viewerHasStarred(String alias) {
        fields.add(new GraphQLResponseField("viewerHasStarred").alias(alias));
        return this;
    }

    public GistResponseProjection typename() {
        return typename(null);
    }

    public GistResponseProjection typename(String alias) {
        fields.add(new GraphQLResponseField("__typename").alias(alias));
        return this;
    }


}

File: src/main/kotlin-graphql/io/github/graphql/model/GistResponseProjection.kt
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection

/**
 * Response projection for Gist
 */
@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-06T19:48:03+0800"
)
open class GistResponseProjection : GraphQLResponseProjection() {

    override fun `all$`(): GistResponseProjection = `all$`(3)

    override fun `all$`(maxDepth: Int): GistResponseProjection {
        if (projectionDepthOnFields.getOrDefault("GistResponseProjection.GistCommentConnectionResponseProjection.comments", 0) <= maxDepth) {
            projectionDepthOnFields["GistResponseProjection.GistCommentConnectionResponseProjection.comments"] = projectionDepthOnFields.getOrDefault("GistResponseProjection.GistCommentConnectionResponseProjection.comments", 0) + 1
            this.comments(GistCommentConnectionResponseProjection().`all$`(maxDepth - projectionDepthOnFields.getOrDefault("GistResponseProjection.GistCommentConnectionResponseProjection.comments", 0)))
        }
        this.createdAt()
        this.description()
        if (projectionDepthOnFields.getOrDefault("GistResponseProjection.GistFileResponseProjection.files", 0) <= maxDepth) {
            projectionDepthOnFields["GistResponseProjection.GistFileResponseProjection.files"] = projectionDepthOnFields.getOrDefault("GistResponseProjection.GistFileResponseProjection.files", 0) + 1
            this.files(GistFileResponseProjection().`all$`(maxDepth - projectionDepthOnFields.getOrDefault("GistResponseProjection.GistFileResponseProjection.files", 0)))
        }
        if (projectionDepthOnFields.getOrDefault("GistResponseProjection.GistConnectionResponseProjection.forks", 0) <= maxDepth) {
            projectionDepthOnFields["GistResponseProjection.GistConnectionResponseProjection.forks"] = projectionDepthOnFields.getOrDefault("GistResponseProjection.GistConnectionResponseProjection.forks", 0) + 1
            this.forks(GistConnectionResponseProjection().`all$`(maxDepth - projectionDepthOnFields.getOrDefault("GistResponseProjection.GistConnectionResponseProjection.forks", 0)))
        }
        this.id()
        this.isFork()
        this.isPublic()
        this.name()
        if (projectionDepthOnFields.getOrDefault("GistResponseProjection.RepositoryOwnerResponseProjection.owner", 0) <= maxDepth) {
            projectionDepthOnFields["GistResponseProjection.RepositoryOwnerResponseProjection.owner"] = projectionDepthOnFields.getOrDefault("GistResponseProjection.RepositoryOwnerResponseProjection.owner", 0) + 1
            this.owner(RepositoryOwnerResponseProjection().`all$`(maxDepth - projectionDepthOnFields.getOrDefault("GistResponseProjection.RepositoryOwnerResponseProjection.owner", 0)))
        }
        this.pushedAt()
        this.resourcePath()
        if (projectionDepthOnFields.getOrDefault("GistResponseProjection.StargazerConnectionResponseProjection.stargazers", 0) <= maxDepth) {
            projectionDepthOnFields["GistResponseProjection.StargazerConnectionResponseProjection.stargazers"] = projectionDepthOnFields.getOrDefault("GistResponseProjection.StargazerConnectionResponseProjection.stargazers", 0) + 1
            this.stargazers(StargazerConnectionResponseProjection().`all$`(maxDepth - projectionDepthOnFields.getOrDefault("GistResponseProjection.StargazerConnectionResponseProjection.stargazers", 0)))
        }
        this.updatedAt()
        this.url()
        this.viewerHasStarred()
        this.typename()
        return this
    }

    fun comments(subProjection: GistCommentConnectionResponseProjection): GistResponseProjection = comments(null, subProjection)

    fun comments(alias: String?, subProjection: GistCommentConnectionResponseProjection): GistResponseProjection {
        fields.add(GraphQLResponseField("comments").alias(alias).projection(subProjection))
        return this
    }

    fun comments(input: GistCommentsParametrizedInput, subProjection: GistCommentConnectionResponseProjection): GistResponseProjection = comments(null, input, subProjection)

    fun comments(alias: String?, input: GistCommentsParametrizedInput, subProjection: GistCommentConnectionResponseProjection): GistResponseProjection {
        fields.add(GraphQLResponseField("comments").alias(alias).parameters(input).projection(subProjection))
        return this
    }

    fun createdAt(): GistResponseProjection = createdAt(null)

    fun createdAt(alias: String?): GistResponseProjection {
        fields.add(GraphQLResponseField("createdAt").alias(alias))
        return this
    }

    fun description(): GistResponseProjection = description(null)

    fun description(alias: String?): GistResponseProjection {
        fields.add(GraphQLResponseField("description").alias(alias))
        return this
    }

    fun files(subProjection: GistFileResponseProjection): GistResponseProjection = files(null, subProjection)

    fun files(alias: String?, subProjection: GistFileResponseProjection): GistResponseProjection {
        fields.add(GraphQLResponseField("files").alias(alias).projection(subProjection))
        return this
    }

    fun files(input: GistFilesParametrizedInput, subProjection: GistFileResponseProjection): GistResponseProjection = files(null, input, subProjection)

    fun files(alias: String?, input: GistFilesParametrizedInput, subProjection: GistFileResponseProjection): GistResponseProjection {
        fields.add(GraphQLResponseField("files").alias(alias).parameters(input).projection(subProjection))
        return this
    }

    fun forks(subProjection: GistConnectionResponseProjection): GistResponseProjection = forks(null, subProjection)

    fun forks(alias: String?, subProjection: GistConnectionResponseProjection): GistResponseProjection {
        fields.add(GraphQLResponseField("forks").alias(alias).projection(subProjection))
        return this
    }

    fun forks(input: GistForksParametrizedInput, subProjection: GistConnectionResponseProjection): GistResponseProjection = forks(null, input, subProjection)

    fun forks(alias: String?, input: GistForksParametrizedInput, subProjection: GistConnectionResponseProjection): GistResponseProjection {
        fields.add(GraphQLResponseField("forks").alias(alias).parameters(input).projection(subProjection))
        return this
    }

    fun id(): GistResponseProjection = id(null)

    fun id(alias: String?): GistResponseProjection {
        fields.add(GraphQLResponseField("id").alias(alias))
        return this
    }

    fun isFork(): GistResponseProjection = isFork(null)

    fun isFork(alias: String?): GistResponseProjection {
        fields.add(GraphQLResponseField("isFork").alias(alias))
        return this
    }

    fun isPublic(): GistResponseProjection = isPublic(null)

    fun isPublic(alias: String?): GistResponseProjection {
        fields.add(GraphQLResponseField("isPublic").alias(alias))
        return this
    }

    fun name(): GistResponseProjection = name(null)

    fun name(alias: String?): GistResponseProjection {
        fields.add(GraphQLResponseField("name").alias(alias))
        return this
    }

    fun owner(subProjection: RepositoryOwnerResponseProjection): GistResponseProjection = owner(null, subProjection)

    fun owner(alias: String?, subProjection: RepositoryOwnerResponseProjection): GistResponseProjection {
        fields.add(GraphQLResponseField("owner").alias(alias).projection(subProjection))
        return this
    }

    fun pushedAt(): GistResponseProjection = pushedAt(null)

    fun pushedAt(alias: String?): GistResponseProjection {
        fields.add(GraphQLResponseField("pushedAt").alias(alias))
        return this
    }

    fun resourcePath(): GistResponseProjection = resourcePath(null)

    fun resourcePath(alias: String?): GistResponseProjection {
        fields.add(GraphQLResponseField("resourcePath").alias(alias))
        return this
    }

    fun stargazers(subProjection: StargazerConnectionResponseProjection): GistResponseProjection = stargazers(null, subProjection)

    fun stargazers(alias: String?, subProjection: StargazerConnectionResponseProjection): GistResponseProjection {
        fields.add(GraphQLResponseField("stargazers").alias(alias).projection(subProjection))
        return this
    }

    fun stargazers(input: GistStargazersParametrizedInput, subProjection: StargazerConnectionResponseProjection): GistResponseProjection = stargazers(null, input, subProjection)

    fun stargazers(alias: String?, input: GistStargazersParametrizedInput, subProjection: StargazerConnectionResponseProjection): GistResponseProjection {
        fields.add(GraphQLResponseField("stargazers").alias(alias).parameters(input).projection(subProjection))
        return this
    }

    fun updatedAt(): GistResponseProjection = updatedAt(null)

    fun updatedAt(alias: String?): GistResponseProjection {
        fields.add(GraphQLResponseField("updatedAt").alias(alias))
        return this
    }

    fun url(): GistResponseProjection = url(null)

    fun url(alias: String?): GistResponseProjection {
        fields.add(GraphQLResponseField("url").alias(alias))
        return this
    }

    fun viewerHasStarred(): GistResponseProjection = viewerHasStarred(null)

    fun viewerHasStarred(alias: String?): GistResponseProjection {
        fields.add(GraphQLResponseField("viewerHasStarred").alias(alias))
        return this
    }

    fun typename(): GistResponseProjection = typename(null)

    fun typename(alias: String?): GistResponseProjection {
        fields.add(GraphQLResponseField("__typename").alias(alias))
        return this
    }


}
